src/azure-cli/azure/cli/command_modules/eventgrid/custom.py

```diff
@@ -1369,37 +1428,88 @@ def _update_event_subscription_internal(  # pylint: disable=too-many-locals,too-
     return params
 
 
+def _validate_destination_attribute(endpoint_type, storage_queue_msg_ttl=None, delivery_attribute_mapping=None):
+    isStorageQueueDestination = endpoint_type is not None and endpoint_type.lower() == STORAGEQUEUE_DESTINATION.lower()
+
+    if not isStorageQueueDestination and storage_queue_msg_ttl is not None:
+        raise CLIError('usage error: --storage-queue-msg-ttl is only applicable for endpoint type StorageQueue.')
+
+    if isStorageQueueDestination and delivery_attribute_mapping is not None:
+        raise CLIError('usage error: --delivery-attribute-mapping is not applicable for endpoint type StorageQueue.')
+
+
+def _set_event_subscription_destination(
+        destination,
+        storage_queue_msg_ttl=None,
+        delivery_attribute_mapping=None):
+
+    endpoint_type = destination.endpoint_type
+    if endpoint_type.lower() == WEBHOOK_DESTINATION.lower():
+        if delivery_attribute_mapping is not None:
+            destination.delivery_attribute_mappings = delivery_attribute_mapping
+    elif endpoint_type.lower() == EVENTHUB_DESTINATION.lower():
+        if delivery_attribute_mapping is not None:
+            destination.delivery_attribute_mappings = delivery_attribute_mapping
+    elif endpoint_type.lower() == HYBRIDCONNECTION_DESTINATION.lower():
+        if delivery_attribute_mapping is not None:
+            destination.delivery_attribute_mappings = delivery_attribute_mapping
+    elif endpoint_type.lower() == STORAGEQUEUE_DESTINATION.lower():
+        if storage_queue_msg_ttl is not None:
+            destination.queue_message_time_to_live_in_seconds = storage_queue_msg_ttl
+    elif endpoint_type.lower() == SERVICEBUSQUEUE_DESTINATION.lower():
+        if delivery_attribute_mapping is not None:
+            destination.delivery_attribute_mappings = delivery_attribute_mapping
+    elif endpoint_type.lower() == SERVICEBUSTOPIC_DESTINATION.lower():
+        if delivery_attribute_mapping is not None:
+            destination.delivery_attribute_mappings = delivery_attribute_mapping
+    elif endpoint_type.lower() == AZUREFUNCTION_DESTINATION.lower():
+        if delivery_attribute_mapping is not None:
+            destination.delivery_attribute_mappings = delivery_attribute_mapping
```

Comment:
```suggestion
    if endpoint_type.lower() == STORAGEQUEUE_DESTINATION.lower():
        if storage_queue_msg_ttl is not None:
            destination.queue_message_time_to_live_in_seconds = storage_queue_msg_ttl
    elif delivery_attribute_mapping is not None:
        destination.delivery_attribute_mappings = delivery_attribute_mapping
```

Reply:
same as earlier. it looks right for now but when we add new properties (which happens quite often) it makes it hard to reuse. So leaving it as-is.

Reply:
thank you but it will be harder when we add new properties. so will keep it as-is.

---
In reply to: [603760609](https://github.com/Azure/azure-cli/pull/17496#discussion_r603760609) [](ancestors = 603760609)